### DISCUSSION TITLE
Extension Endpoint Configuration in Monaco

### DISCUSSION THREAD

**Discussion by Aditya-Sinha1990** [Dynatrace/dynatrace-configuration-as-code#476] [Q&A]
Hi,
This is to get some insight on custom extensions in Monaco. Mainly MSSQL, MYSQL Pre-Defined Custom Extension
_Upon using download command for **extension**-_
**Issue1:** Monaco does not fetch the existing extensions endpoints and just a generic template is present in extension Dir of Monaco Directory Structure 

_Upon using deploy command for extension-_
**Issue2:** Configuration complains of **400 location is null**. 
The template does not have the key _location_ and also I hope I don't have to upload the extension zip file as it's already present. I am just trying to add an endpoint configuration in the custom extension.

**Comment by mdicss** [comment]
Hi Aditya-Sinha
I finally managed deploying pmi metrics definitions with monaco. I had to give attention to 'metricGroup' wich in my case has to be 'custom.pmi....' and 'name' which has to be 'custom.pmi....' both followed by the name I gave the extension definition.
Then I defined some metrics and exported their definition in the gui to a json file. This json file could be used as a template to write the monaco configuration togeather with the api documentation for the extension.
